### PR TITLE
Enable support for Vault secrets in Manifester

### PR DESCRIPTION
**Pull request by synkd**
A CI test failure in https://github.com/SatelliteQE/robottelo/pull/14633 revealed that Manifester was unable to interpolate secrets stored in HashiCorp Vault. This PR adds support for Vault secrets based on Robottelo's implementation of Vault integration.

### CODE REVIEW

File: manifester/commands.py [hunk 1]
```diff
@@ -60,8 +60,13 @@ def delete(allocations, all_, remove_manifest_file):
                 uuid=allocation.get("uuid")
             )
             if remove_manifest_file:
+                manifester_directory = (
+                    Path(os.environ["MANIFESTER_DIRECTORY"])
```
Comment (JacobCallahan) [inline manifester/commands.py:64]: Might want to slap on a resolve here
```suggestion
                    Path(os.environ["MANIFESTER_DIRECTORY"]).resolve()
```

File: manifester/helpers.py [hunk 2]
```diff
@@ -226,3 +235,139 @@ def __getitem__(self, key):
...
+    )
+
+    def __init__(self, env_file=".env"):
+        manifester_directory = Path()
+
+        if "MANIFESTER_DIRECTORY" in os.environ:
+            envar_location = Path(os.environ["MANIFESTER_DIRECTORY"])
+            if envar_location.is_dir():
+                manifester_directory = envar_location
+        self.env_path = manifester_directory.joinpath(env_file)
```
Comment (JacobCallahan) [inline manifester/helpers.py:260]: These would likely be best done in a common place, like the settings module

File: manifester/settings.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 import os
 from pathlib import Path
 
-from dynaconf import Dynaconf, Validator
+from dynaconf import LazySettings, Validator
```
Comment (JacobCallahan) [inline manifester/settings.py:5]: Why this change?

File: manifester/helpers.py [hunk 4]
```diff
@@ -1,15 +1,24 @@
...
 import time
 
 from logzero import logger
 from requests import HTTPError
 import yaml
 
+from manifester.logger import setup_logzero
 from manifester.settings import settings
 
+setup_logzero(level="info")
```
Comment (JacobCallahan) [inline manifester/helpers.py:19]: If switching to using the settings module, you can just use the configured log level

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

File: scripts/vault_login.py
Comment (ogajduse) [inline scripts/vault_login.py:1]: Looking at https://hvac.readthedocs.io/en/stable/usage/auth_methods/jwt-oidc.html#oidc-authorization-url-request, it should be possible to use pure Python implementation, using `hvac`, of vault login without needing to install `vault` CLI tool. It is something I have wanted to explore some time ago, but I never found time for it.
If you would like to dig into it, I would be happy to assist/help you.

**Comment by jyejare** [comment]
@synkd What if we would have :

1. Read the secret using robottelo's settings object.
2. Pass the secret value to manifester wherever needed.

Hence we don't need to rewrite the code in manifester for vault integration.

Manifester being a standalone tool and since we are the only users of it, does not need to store/access the secrets from vault and for now its just a one secret.

File: manifester/settings.py (hunk 3 above)
Comment (synkd) [reply manifester/settings.py:5]: This change was part of my initial debugging. In trying to determine why my Vault integration was failing, I tried matching the Robottelo implementation as closely as possible. I can revert this.

**Comment by synkd** [comment]
@jyejare I have been trying to avoid having Manifester depend directly on Robottelo. Although SatQE is currently the only group using Manifester, I think it has potential applications elsewhere as well. However, I'm open to discussion on the topic. Maybe we can discuss this at the Automation Brainstorming meeting in two weeks, along with Ondrej's suggestion above?

File: manifester/helpers.py (hunk 2 above)
Comment (synkd) [reply manifester/helpers.py:260]: @JacobCallahan This actually is done in the settings module, but I revisited this today and remembered why I had repeated it here: vault_login.py throws an error when trying to retrieve settings values if the user is not already logged in to Vault. If I'm understanding the situation correctly, vault_login.py imports helpers.py, in which it encounters an assignment that uses settings.get(), then dynaconf attempts to process the settings file, which includes a Vault-formatted setting (offline_token), and dynaconf subsequently tries to retrieve that value from Vault before the login process has completed. This is also why I had hard-coded the log level in this module, which you commented on as well.

There may be another way to work around this race condition, but I have not figured one out yet. Do you have any thoughts on an alternative, or alternatively, do you think my admittedly inelegant solution is acceptable?

File: manifester/settings.py (hunk 3 above)
Comment (JacobCallahan) [reply manifester/settings.py:5]: Yeah they're equivalent statements since `Dynaconf` is an alias to `LazySettings`. However, it's recommended to use `Dynaconf` here instead. Robottelo is likely just holding on to an old pattern (and should be changed).

https://github.com/dynaconf/dynaconf/blob/master/dynaconf/__init__.py#L28

File: manifester/helpers.py (hunk 2 above)
Comment (JacobCallahan) [reply manifester/helpers.py:260]: Hmmm not ideal, but if it needs to be repeated anyway for the vault login then it's up to you whether that repetition happens here or in the vault script. I have no objections either way now.

**Review by JacobCallahan — APPROVED** [review]
No major blocking recommendations remaining for me. However, I will say that manifester should never have robottelo as its dependency. This would result in a circular dependency between the two.
